payload_test.go

```diff
@@ -0,0 +1,102 @@
+package bugsnag
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/bugsnag/bugsnag-go/errors"
+)
+
+const expSmall = `{"apiKey":"","events":[{"app":{"releaseStage":""},"device":{},"exceptions":[{"errorClass":"","message":"","stacktrace":null}],"metaData":{},"payloadVersion":"4","severity":"","unhandled":false}],"notifier":{"name":"Bugsnag Go","url":"https://github.com/bugsnag/bugsnag-go","version":"1.3.1"}}`
+
+// The large payload has a timestamp in it which makes it awkward to assert against.
+// Instead, assert that the timestamp property exist, along with the rest of the expected payload
+const expLargePre = `{"apiKey":"166f5ad3590596f9aa8d601ea89af845","events":[{"app":{"releaseStage":"mega-production","type":"gin","version":"1.5.2"},"context":"/api/v2/albums","device":{"hostname":"super.duper.site"},"exceptions":[{"errorClass":"error class","message":"error message goes here","stacktrace":[{"method":"doA","file":"a.go","lineNumber":65},{"method":"fetchB","file":"b.go","lineNumber":99,"inProject":true},{"method":"incrementI","file":"i.go","lineNumber":651}]}],"groupingHash":"custom grouping hash","metaData":{"custom tab":{"my key":"my value"}},"payloadVersion":"4","session":{"startedAt":"`
+const expLargePost = `,"severity":"info","severityReason":{"attributes":{"framework":"gin"},"type":"unhandledError"},"unhandled":true,"user":{"id":"1234baerg134","name":"Kool Kidz on da bus","email":"typo@busgang.com"}}],"notifier":{"name":"Bugsnag Go","url":"https://github.com/bugsnag/bugsnag-go","version":"1.3.1"}}`
+
+func TestMarshalEmptyPayload(t *testing.T) {
+	payload := payload{&Event{}, &Configuration{}}
+	bytes, _ := payload.MarshalJSON()
+	if got := string(bytes[:]); got != expSmall {
+		t.Errorf("Payload different to what was expected. \nGot: %s\nExp: %s", got, expSmall)
+	}
+}
+
+func TestMarshalLargePayload(t *testing.T) {
+	payload := makeLargePayload()
+	bytes, _ := payload.MarshalJSON()
+	got := string(bytes[:])
+	if !strings.Contains(got, expLargePre) {
+		t.Errorf("Expected large payload to contain\n'%s'\n but was\n'%s'", expLargePre, got)
+	}
+	if !strings.Contains(got, expLargePost) {
+		t.Errorf("Expected large payload to contain\n'%s'\n but was\n'%s'", expLargePost, got)
+	}
+}
+
+func BenchmarkEmptyMarshalJSON(b *testing.B) {
```

Comment:
We don't really care about keeping these going forward so should delete. They're also incorrect as the compiler is likely to optimise away some of this as it currently stands.